test/wowman/core_test.clj

```diff
@@ -139,38 +139,43 @@
         (core/stop app-state)))))
 
 (deftest paths
-  (let [app-state (core/start {})]
-    (try
-      (testing "all path keys are using a known suffix"
-        (doseq [key (keys (core/paths))]
-          (is (some #{"dir" "file" "uri"} (clojure.string/split (name key) #"\-")))))
-
-      (testing "all paths to files and directories are absolute"
-        (let [files+dirs (filter (fn [[k v]] (or (ends-with? k "-dir")
-                                                 (ends-with? k "-file")))
-                                 (core/paths))]
-          (doseq [[key path] files+dirs]
-            (is (-> path (starts-with? "/")) (format "path %s is not absolute: %s" key path)))))
-
-      (testing "all remote paths are using https"
-        (let [remote-paths (filter (fn [[k v]] (ends-with? k "-uri")) (core/paths))]
-          (doseq [[key path] remote-paths]
-            (is (-> path (starts-with? "https://")) (format "remote path %s is not using HTTPS: %s" key path)))))
-
-      (comment
-        "what exactly am I testing here again? this path overriding has plenty of coverage already. see test_helper.clj"
-        (testing "XDG data paths can be overridden with environment variables"
-          (with-env [:xdg-data-home "/foo", :xdg-config-home "/bar"]
-            (is (= "/foo" (:data-dir (core/paths))))
-            (is (= "/bar" (:config-dir (core/paths)))))))
-
-      (testing "paths that cannot be written to raise a runtime exception"
-        (with-env [:xdg-data-home "/foo", :xdg-config-home "/bar"]
-          (core/stop app-state)
-          (is (thrown? RuntimeException (core/start {})))))
-
-      (finally
-        (core/stop app-state)))))
+  (with-running-app
+    (testing "all path keys are using a known suffix"
+      (doseq [key (keys (core/paths))]
+        (is (some #{"dir" "file" "uri"} (clojure.string/split (name key) #"\-")))))
+
+    (testing "all paths to files and directories are absolute"
+      (let [files+dirs (filter (fn [[k v]] (or (ends-with? k "-dir")
```

Comment:
is this necessary? I think there may be a runtime check on paths that ensures everything has a proper suffix

Reply:
derp, ignore me